framework/include/restart/DataIO.h

```diff
@@ -256,6 +269,27 @@ dataStore(std::ostream & stream, std::map<T,U> & m, void * context)
 
 template<typename T, typename U>
 inline void
+dataStore(std::ostream & stream, std::unordered_map<T,U> & m, void * context)
+{
+  // First store the size of the map
+  unsigned int size = m.size();
+  stream.write((char *) &size, sizeof(size));
+
+  typename std::unordered_map<T,U>::iterator it = m.begin();
+  typename std::unordered_map<T,U>::iterator end = m.end();
+
+  for (; it != end; ++it)
```

Comment:
That explicit for loop is used throughout this file for maps. Do you suggest changing all them?